Added VPC Quickstart DA with predefined ACL profiles and optional VPC Flow Logs (#1082)
terraform-ibm-modules/terraform-ibm-landing-zone-vpc: reference-architectures/deploy-arch-ibm-vpc-quickstart-vpc.md

---

copyright:
  years: 2025
lastupdated: "2025-12-02"

keywords:

subcollection: deployable-reference-architectures

authors:
  - name: "Khuzaima Shakeel"

# The release that the reference architecture describes

> version: 1.0.0

Author: Done

## Requirements
{: #ra-vpc-quickstart-components}

### The following table outlines the requirements that are addressed in this architecture.

| Requirement | Component | Reasons for choice | Alternative choice |
|------------|-----------|--------------------|--------------------|
| *Provide a basic, ready-to-use VPC with minimal inputs* | Predefined VPC | Deploys a VPC quickly without requiring users to design networking components | Use the fully configurable variation for granular control |
| *Create availability-zone redundancy* | Fixed three-zone subnets | Ensures high availability by provisioning one subnet per zone automatically | Use the fully configurable variation for flexibility |
| *Basic traffic governance* | Network profile selector (open, standard, ibm-internal, closed) | Provides simple, predefined ACL behavior without requiring custom rules | Define custom ACL rules and SG rules manually in the fully configurable variation |

{: caption="QuickStart variation of Cloud foundation for VPC" caption-side="bottom"}

---

## Key features
{: #ra-vpc-quickstart-features}

### VPC Setup
- Automatically creates a new VPC with IBM-recommended defaults

> - Deploys **one subnet per zone** (three total)

Author: Done

### Built-in Network Profiles
- **Open** – Public & Open Access

> - **Standard** – Internet & Internal Access (Default)

Reviewer: Internet access - gives an impression for an open network access. Though you are opening few ports only, so please update it accordingly.

Author: Done

- **IBM Internal** – Private Network Only

> - **Closed** – Fully Isolated

Author: Done

### Security & Network Defaults
- ACLs applied according to selected network profile
- Security group automatically cleaned for restrictive profiles
- Public gateways created only when allowed by the profile

### Optional Flow Logs

> - Enable VPC Flow Logs to create a Object Storage instance and bucket automatically

Reviewer: ```suggestion
### Flow Logs
- VPC Flow Logs can be enabled if needed.
```

> - VPC Flow Logs can be enabled if needed.

Reviewer 2: Mention about the creation of COS resources if its enabled

{: caption="QuickStart variation of Cloud foundation for VPC" caption-side="bottom"}

# Key features
{: #ra-vpc-quickstart-features}

## VPC Setup
- Automatically creates a new VPC with IBM-recommended defaults
- Sets up one subnet per zone, resulting in three subnets.

## Built-in Network Profiles

The following network profiles provide predefined security postures by configuring **Network ACLs**, **public gateway access**, and **default security group behavior**. These profiles align exactly with the options exposed in the IBM Cloud catalog UI.

- **Open (Public & Open Access)**
  Allows all inbound and outbound traffic. Suitable for testing or unrestricted workloads.

- **Standard (Specific Ports & Internal Access)** *(Default)*
  Allows traffic on common service ports (SSH 22, HTTP 80, HTTPS 443), enables IBM Cloud internal connectivity.

- **IBM Cloud private backbone (Private Network Only)**
  No external/public connectivity. Only IBM internal and VPC connectivity allowed. Intended for internal-only workloads that must not be exposed publicly.
  Learn more: https://cloud.ibm.com/docs/vpc?topic=vpc-about-networking#private-network

- **Closed (Fully Isolated)**

>   Fully locked-down environment with no inbound or outbound connectivity. Suitable for highly sensitive or isolated security scenarios.

Author: Done

    description: "A deployable architecture that provides full control over VPC networking, security, and connectivity components."
  - title: "Cloud foundation for VPC (Standard - Financial Services edition)"
    url: "https://cloud.ibm.com/docs/deployable-reference-architectures?topic=deployable-reference-architectures-vpc-ra"
    description: "A VPC architecture based on IBM Cloud for Financial Services controls."

use-case: Foundational Infrastructure
compliance: None

content-type: reference-architecture

---

{{site.data.keyword.attribute-definition-list}}

# Cloud foundation for VPC – QuickStart (Basic and simple)
{: #vpc-quickstart-ra}
{: toc-content-type="reference-architecture"}
{: toc-industry="CrossIndustry"}
{: toc-use-case="Foundational Infrastructure"}
{: toc-version="1.0.0"}

The QuickStart variation provides a **simple and beginner-friendly** Virtual Private Cloud (VPC) deployment that requires minimal configuration. It helps users rapidly create a functional network environment on IBM Cloud without needing to define subnets, ACL rules, security groups, or connectivity patterns manually.

> This variation is ideal for users who want a **basic VPC setup**, lightweight networking defaults, and the option to enable VPC Flow Logs.

Reviewer: ```suggestion
The QuickStart variation of the Cloud foundation for VPC provides a **basic and simple** Virtual Private Cloud (VPC) deployment that requires minimal configuration. It enables users to quickly create a functional network environment on IBM Cloud. This variation is best suited for users who need a **basic VPC configuration** with lightweight networking defaults and support for VPC Flow Logs.
```

---

## Architecture diagram
{: #ra-vpc-quickstart-architecture}

> ![Architecture diagram for the QuickStart variation of Cloud foundation for VPC](deployable-architecture-quickstart-vpc.svg "QuickStart VPC architecture"){: caption="QuickStart VPC architecture" caption-side="bottom"}{: external download="deployable-architecture-quickstart-vpc.svg"}

Author: Done

{: #ra-vpc-quickstart-components-arch}

| Requirement | Component | Reasons for choice | Alternative |
|------------|-----------|--------------------|-------------|
| *Provide a basic, ready-to-use VPC with minimal inputs* | Predefined VPC | Deploys a VPC quickly without requiring the user to design the network | Use the fully configurable variation for deeper customization |
| *Create availability-zone redundancy* | Fixed three-zone subnets | One subnet per zone to ensure multi-AZ coverage | Single-zone deployment (not recommended) |
| *Basic traffic governance* | Network profile selector (open, standard, ibm-internal, closed) | Users can choose the desired security posture without written ACL rules | Manually writing custom ACLs |

{: caption="VPC architecture decisions" caption-side="bottom"}

---

### Networking and connectivity decisions
{: #ra-vpc-quickstart-components-connectivity}

| Requirement | Component | Reason | Alternative |
|------------|-----------|--------|-------------|
| *Optional access to the internet* | Public gateways per zone (automatic) | Enabled only for `open` and `standard` profiles | No public gateways for locked-down profiles |

> | *Subnet-level traffic control* | Network ACL profiles | Simplifies security without requiring rule definitions | Fully customizable ACLs (in advanced variation) |

Author: Done

> | *Instance-level default security* | Default VPC security group | Automatically cleaned when selecting restrictive profiles (`ibm-internal`, `closed`) | Custom security group rules |

Author: removed

{: caption="Networking and connectivity decisions" caption-side="bottom"}

---

### Simplicity and user experience decisions
{: #ra-vpc-quickstart-components-simplicity}

| Requirement | Component | Reasons | Alternative |
|------------|-----------|---------|-------------|

> | *Zero-effort deployment* | Predefined subnets + ACL mapping | Users only pick prefix, region, and profile | Manual subnet planning |

Reviewer: `Users only pick prefix, region, and profile` -> `Provide only prefix, region and network_profile`
Something like this...if it is technically correct

> | *Zero-effort deployment* | Predefined subnets + ACL mapping | Users only pick prefix, region, and profile | Manual subnet planning |

Author: Done

| *Clear security posture options* | User-friendly “Network Profile” widget | Shows descriptions and recommendations | Plain text input |

> | *Simple observability integration* | Optional VPC Flow Logs → COS | Enabled via toggle | External log collectors |

Reviewer: ```suggestion
| *Observability integration* | VPC Flow Logs  | Enabled via toggle | External log collectors |
```

{: caption="Simplicity decision points" caption-side="bottom"}

---

## Key features
{: #ra-vpc-quickstart-features}

### Core VPC Setup
- Automatically creates a new VPC with IBM-recommended defaults
- Deploys **one subnet per zone** (three total)

### Built-in Network Profiles
- **Open** – Unrestricted
- **Standard** – SSH/HTTP/HTTPS + IBM internal rules
- **IBM Internal** – No inbound customer traffic

> - **Closed** – Fully restricted

Author: Done

### Security & Network Defaults
- ACLs applied according to selected network profile
- Security group automatically cleaned for restrictive profiles
- Public gateways created only when allowed by the profile

### Optional Flow Logs

> - Enable VPC Flow Logs to create a COS instance and bucket automatically

Reviewer: Please use Object Storage instead of COS

Author: Done
